--- mono-4.8.1/scripts/mono-find-provides.in~	2017-04-12 12:51:22.000000000 +0200
+++ mono-4.8.1/scripts/mono-find-provides.in	2018-10-21 19:25:53.753471755 +0200
@@ -16,7 +16,7 @@
 
 # Only include files with /gac/ in path
 #  (Allows packages to contain private assemblies that don't conflict with other packages)
-monolist=($(printf "%s\n" "${monolist[@]}" | egrep "/gac/"))
+#monolist=($(printf "%s\n" "${monolist[@]}" | egrep "/gac/"))
 # Disabled... see ChangeLog
 
 # Set the prefix, unless it is overriden (used when building mono rpms)
