--- mono-4.8.1/scripts/mono-find-requires.in~	2018-10-21 17:31:32.000000000 +0200
+++ mono-4.8.1/scripts/mono-find-requires.in	2018-10-21 17:33:52.246228263 +0200
@@ -15,6 +15,10 @@
 filelist=($(grep -Ev '/usr/doc/|/usr/share/doc/'))
 monolist=($(printf "%s\n" "${filelist[@]}" | egrep "\\.(exe|dll)\$"))
 
+# Only include files with /gac/ in path
+#  (Allows packages to contain private assemblies that don't conflict with other packages)
+monolist=($(printf "%s\n" "${monolist[@]}" | egrep "/gac/"))
+
 # parse .config files to find which native libraries to depend on 
 #  (target attribute must have double quotes for this to work, ie: target="file" )
 # Add /etc/mono/config ?
