--- mono-3.1.2/mono/mini/Makefile.am.orig	2013-07-17 02:33:58.000000000 +0200
+++ mono-3.1.2/mono/mini/Makefile.am	2013-07-21 21:05:53.293560192 +0200
@@ -118,7 +118,7 @@
 	ln -sf $< $@
 
 install-exec-hook:
-	ln -sf $(DESTDIR)$(bindir)/mono-$(mono_bin_suffix) $(DESTDIR)$(bindir)/mono
+	ln -sf mono-$(mono_bin_suffix) $(DESTDIR)$(bindir)/mono
 	(cd $(DESTDIR)$(libdir); for i in libmono$(libmono_suffix)*; do ln -sf $$i `echo $$i | sed s/$(libmono_suffix)//` ; done)
 endif
 
