--- mcs/build/library.make	(revision 38861)
+++ mcs/build/library.make	(working copy)
@@ -87,8 +87,6 @@
 SNFLAGS = -q -R
 endif
 
-libdir = $(prefix)/lib
-
 ifeq ($(PLATFORM), win32)
 GACDIR = `cygpath -w $(libdir)`
 GACROOT = `cygpath -w $(DESTDIR)$(libdir)`
--- mcs/build/config-default.make	(revision 38861)
+++ mcs/build/config-default.make	(working copy)
@@ -14,6 +14,8 @@
 CFLAGS = -g -O2
 INSTALL = /usr/bin/install
 prefix = /usr/local
+exec_prefix = $(prefix)
+libdir = $(prefix)/lib
 RUNTIME = mono $(RUNTIME_FLAGS)
 TEST_RUNTIME = MONO_PATH="$(topdir)/class/lib/$(PROFILE)$(PLATFORM_PATH_SEPARATOR)$(TEST_MONO_PATH)$(PLATFORM_PATH_SEPARATOR)$$MONO_PATH" $(RUNTIME) --debug
 
--- mcs/build/executable.make	(revision 38861)
+++ mcs/build/executable.make	(working copy)
@@ -31,7 +31,7 @@
 else
 
 ifndef PROGRAM_INSTALL_DIR
-PROGRAM_INSTALL_DIR = $(prefix)/lib/mono/$(FRAMEWORK_VERSION)
+PROGRAM_INSTALL_DIR = $(libdir)/mono/$(FRAMEWORK_VERSION)
 endif
 
 install-local: $(PROGRAM) $(PROGRAM_config)
--- configure.in	(revision 39087)
+++ configure.in	(working copy)
@@ -1630,6 +1630,8 @@
   esac
 
   echo "prefix=$exec_prefix" > $srcdir/$mcsdir/build/config.make
+  echo "exec_prefix=$exec_prefix" >> $srcdir/$mcsdir/build/config.make
+  echo "libdir=$libdir" >> $srcdir/$mcsdir/build/config.make
   echo 'MCS_FLAGS = $(PLATFORM_DEBUG_FLAGS)' >> $srcdir/$mcsdir/build/config.make
 )
 
--- mcs/tools/corcompare/Makefile	(revision 38861)
+++ mcs/tools/corcompare/Makefile	(working copy)
@@ -21,7 +21,7 @@
 	MissingType.cs \
 	ToDoAssembly.cs
 
-PROGRAM_INSTALL_DIR = $(prefix)/lib/mono/$(FRAMEWORK_VERSION)
+PROGRAM_INSTALL_DIR = $(libdir)/mono/$(FRAMEWORK_VERSION)
 
 APIINFO_SOURCES = mono-api-info.cs
 APIDIFF_SOURCES = mono-api-diff.cs
--- mcs/tools/security/Makefile	(revision 38861)
+++ mcs/tools/security/Makefile	(working copy)
@@ -8,7 +8,7 @@
 SECURITY_PROGRAMS = secutil.exe cert2spc.exe sn.exe MakeCert.exe chktrust.exe signcode.exe setreg.exe certmgr.exe caspol.exe permview.exe
 SECURITY_SOURCES = AssemblyInfo.cs StrongNameManager.cs $(SECURITY_PROGRAMS:.exe=.cs)
 
-PROGRAM_INSTALL_DIR = $(prefix)/lib/mono/$(FRAMEWORK_VERSION)
+PROGRAM_INSTALL_DIR = $(libdir)/mono/$(FRAMEWORK_VERSION)
 
 DISTFILES = README $(SECURITY_SOURCES)
 
--- mcs/class/corlib/Makefile	(revision 38861)
+++ mcs/class/corlib/Makefile	(working copy)
@@ -19,7 +19,7 @@
 # disable, until people fix their code...
 #corlib_flags = /unsafe /nostdlib /d:INSIDE_CORLIB
 
-LIBRARY_INSTALL_DIR = $(prefix)/lib/mono/$(FRAMEWORK_VERSION)
+LIBRARY_INSTALL_DIR = $(libdir)/mono/$(FRAMEWORK_VERSION)
 
 CLEAN_FILES = $(cmplib) $(reslib) $(plattestlib) $(plattestlib).sources \
 	       $(cmp_response) $(cmp_makefrag) \
