--- mono-0.16/mono/jit/Makefile.am.wiget	Thu Nov 28 17:29:44 2002
+++ mono-0.16/mono/jit/Makefile.am	Thu Nov 28 17:31:47 2002
@@ -48,7 +48,7 @@
 	debug.h			\
 	regset.h
 
-mono_SOURCES = mono.c $(jitsources)
+mono_SOURCES = mono.c 
 
 #mono_LDFLAGS=-static
 mono_LDADD =				\
@@ -59,6 +59,7 @@
 	../os/libmonoos.la		\
 	$(GLIB_LIBS)			\
 	$(GMODULE_LIBS) 		\
+	libmono.la			\
 	-lm
 
 codegen-x86.c codegen.h: x86.brg
