diff -ruN mono-1.1.13.6./data/config.in mono-1.1.13.6/data/config.in
--- mono-1.1.13.6./data/config.in	2006-03-29 17:34:40.000000000 +0200
+++ mono-1.1.13.6/data/config.in	2006-03-29 17:46:04.827782344 +0200
@@ -2,9 +2,9 @@
 	<dllmap dll="cygwin1.dll" target="@LIBC@" />
 	<dllmap dll="libc" target="@LIBC@" />
 	<dllmap dll="intl" target="@INTL@" />
-	<dllmap dll="libxslt.dll" target="libxslt@libsuffix@" />
-	<dllmap dll="libmySQL.dll" target="libmysqlclient@libsuffix@" />
-	<dllmap dll="odbc32.dll" target="libodbc@libsuffix@" />
+	<dllmap dll="libxslt.dll" target="libxslt.so.1" />
+	<dllmap dll="libmySQL.dll" target="libmysqlclient.so.15" />
+	<dllmap dll="odbc32.dll" target="libodbc.so.1" />
 	<dllmap dll="oci" target="libclntsh@libsuffix@" />
 	<dllmap dll="db2cli" target="libdb2_36@libsuffix@"/>
 	<dllmap dll="msvcrt" target="@LIBC@"/>
