diff -ur mono-1.1.16/data/config.in mono-1.1.16-patch/data/config.in
--- mono-1.1.16/data/config.in	2006-07-05 23:55:55.000000000 +0200
+++ mono-1.1.16-patch/data/config.in	2006-07-07 10:34:59.000000000 +0200
@@ -3,9 +3,9 @@
 	<dllmap dll="libc" target="@LIBC@" />
 	<dllmap dll="intl" target="@INTL@" />
 	<dllmap dll="libintl" target="@INTL@" />
-	<dllmap dll="libxslt.dll" target="libxslt@libsuffix@" />
-	<dllmap dll="libmySQL.dll" target="libmysqlclient@libsuffix@" />
-	<dllmap dll="odbc32.dll" target="libodbc@libsuffix@" />
+	<dllmap dll="libxslt.dll" target="libxslt.so.1" />
+	<dllmap dll="libmySQL.dll" target="libmysqlclient.so.15" />
+	<dllmap dll="odbc32.dll" target="libodbc.so.1" />
 	<dllmap os="osx" dll="odbc32.dll" target="libiodbc.dylib" />
 	<dllmap dll="oci" target="libclntsh@libsuffix@" />
 	<dllmap dll="db2cli" target="libdb2_36@libsuffix@"/>
