diff -ur mono-1.2.3/mcs/class/Managed.Windows.Forms/build-csproj mono-awk/mcs/class/Managed.Windows.Forms/build-csproj
--- mono-1.2.3/mcs/class/Managed.Windows.Forms/build-csproj	2007-01-25 07:19:43.000000000 +0100
+++ mono-awk/mcs/class/Managed.Windows.Forms/build-csproj	2007-02-07 19:03:51.000000000 +0100
@@ -177,7 +177,7 @@
 {
 cat $Resource | while read SRC; do
 SRC=`echo $SRC | $tr '/' '\\\\'`
-SRC=`echo $SRC | sed 's/-resource://' | gawk -F , '{print "                    RelPath = \"" $1 "\"\n                    CustomToolNameSpace = \"" $2 "\""}' | fgrep -v \"\"`
+SRC=`echo $SRC | sed 's/-resource://' | awk -F , '{print "                    RelPath = \"" $1 "\"\n                    CustomToolNameSpace = \"" $2 "\""}' | fgrep -v \"\"`
 
 cat << EOT
                 <File
