--- mono-0.28/configure.in.wiget	2003-10-16 13:49:54.177533865 +0200
+++ mono-0.28/configure.in	2003-10-16 13:50:28.787272055 +0200
@@ -672,10 +672,10 @@
 		arch_target=sparc;
 		ACCESS_UNALIGNED="no"
 		;;
-#       alpha*-*-linux* | alpha*-*-osf*)
-#		TARGET=ALPHA;
-#		ACCESS_UNALIGNED="no"
-#	       ;;
+       alpha*-*-linux* | alpha*-*-osf*)
+		arch_target=alpha;
+		ACCESS_UNALIGNED="no"
+	       ;;
 #	ia64-*-linux* | ia64-*-hpux*)
 #		TARGET=IA64;
 #		arch_target=ia64;
