--- mono-1.1.8.3/libgc/pthread_support.c.orig	2005-06-19 09:59:53.000000000 +0000
+++ mono-1.1.8.3/libgc/pthread_support.c	2005-09-03 21:07:26.422535974 +0000
@@ -561,10 +561,12 @@
  * put it into a separate function.
  */
 #   ifdef __GNUC__
+#   if defined(THREAD_LOCAL_ALLOC) && !defined(DBG_HDRS_ALL)
 static __attribute__((noinline)) unsigned char* get_gc_thread_key_addr GC_PROTO((void))
 {
 	return (unsigned char*)&GC_thread_key;
 }
+#   endif
 
 void GC_push_thread_structures GC_PROTO((void))
 {
