diff -ur mono-1.2.3/data/config.in mono-soname/data/config.in
--- mono-1.2.3/data/config.in	2007-01-25 07:11:31.000000000 +0100
+++ mono-soname/data/config.in	2007-02-07 18:36:17.000000000 +0100
@@ -3,8 +3,8 @@
 	<dllmap dll="libc" target="@LIBC@" os="!windows"/>
 	<dllmap dll="intl" target="@INTL@" os="!windows"/>
 	<dllmap dll="libintl" target="@INTL@" os="!windows"/>
-	<dllmap dll="libxslt.dll" target="libxslt@libsuffix@" os="!windows"/>
-	<dllmap dll="odbc32.dll" target="libodbc@libsuffix@" os="!windows"/>
+	<dllmap dll="libxslt.dll" target="libxslt.so.1" os="!windows"/>
+	<dllmap dll="odbc32.dll" target="libodbc.so.1" os="!windows"/>
 	<dllmap os="osx" dll="odbc32.dll" target="libiodbc.dylib" />
 	<dllmap dll="oci" target="libclntsh@libsuffix@" os="!windows"/>
 	<dllmap dll="db2cli" target="libdb2_36@libsuffix@" os="!windows"/>
