--- mono-0.97/mono/utils/mono-sha1.c.orig	2004-06-21 20:48:52.000000000 +0200
+++ mono-0.97/mono/utils/mono-sha1.c	2004-06-27 02:24:13.000000000 +0200
@@ -70,13 +70,14 @@
 
 #define SHA1HANDSOFF
 
+#include <stdint.h>
 #include <stdio.h>
 #include <string.h>
 #include "mono-digest.h"
 
 
-typedef          long int int32;
-typedef unsigned long int uint32;
+typedef int32_t int32;
+typedef uint32_t uint32;
 
 /* #include <process.h> */	/* prototype for exit() - JHB */
 /* Using return() instead of exit() - SWR */
