commit 5b3f143191d599c1fcca89125793405150c79424
Author: Rodrigo Kumpera <kumpera@gmail.com>
Date:   Sun Apr 22 21:02:59 2012 -0300

    Fix concurrent sweep thread start code to actually work.

diff --git a/mono/metadata/sgen-marksweep.c b/mono/metadata/sgen-marksweep.c
index 15e2767..78bc368 100644
--- a/mono/metadata/sgen-marksweep.c
+++ b/mono/metadata/sgen-marksweep.c
@@ -1960,7 +1960,7 @@ static void
 post_param_init (void)
 {
 	if (concurrent_sweep) {
-		if (mono_native_thread_create (&ms_sweep_thread, ms_sweep_thread_func, NULL)) {
+		if (!mono_native_thread_create (&ms_sweep_thread, ms_sweep_thread_func, NULL)) {
 			fprintf (stderr, "Error: Could not create sweep thread.\n");
 			exit (1);
 		}
