diff -Nur mono-0.28.orig/mono/arch/sparc/sparc-codegen.h mono-0.28/mono/arch/sparc/sparc-codegen.h
--- mono-0.28.orig/mono/arch/sparc/sparc-codegen.h	2003-08-24 22:49:45.000000000 +0000
+++ mono-0.28/mono/arch/sparc/sparc-codegen.h	2003-10-16 08:39:23.000000000 +0000
@@ -409,7 +409,7 @@
 
 #define sparc_fsubs(ins, r1, r2, dest) sparc_fop( ins, r1, sparc_fsubs_val, r2, dest ) 
 #define sparc_fsubd(ins, r1, r2, dest) sparc_fop( ins, r1, sparc_fsubd_val, r2, dest ) 
-b#define sparc_fsubq(ins, r1, r2, dest) sparc_fop( ins, r1, sparc_fsubq_val, r2, dest ) 
+#define sparc_fsubq(ins, r1, r2, dest) sparc_fop( ins, r1, sparc_fsubq_val, r2, dest ) 
 
 #define sparc_fmuls( ins, r1, r2, dest ) sparc_fop( ins, r1, sparc_fmuls_val, r2, dest )
 #define sparc_fmuld( ins, r1, r2, dest ) sparc_fop( ins, r1, sparc_fmuld_val, r2, dest )
