--- mono-1.0.6/mcs-1.0.6/tools/gacutil/driver.cs.orig	2005-02-24 22:52:33.000000000 +0100
+++ mono-1.0.6/mcs-1.0.6/tools/gacutil/driver.cs	2005-02-24 22:52:55.000000000 +0100
@@ -598,9 +598,7 @@
 		private static string EnsureLib (string dir)
 		{
 			DirectoryInfo d = new DirectoryInfo (dir);
-			if (d.Name == "lib")
 				return dir;
-			return Path.Combine (dir, "lib");
 		}
 
 		private static void WriteLine ()
